lib/backbone.marionette.esm.js

```diff
@@ -846,18 +1235,204 @@ var UIMixin = {
+// 'click:foo'
+
+function buildViewTrigger(view, triggerDef) {
+  if (isString(triggerDef)) {
+    triggerDef = {
+      event: triggerDef
+    };
+  }
+
+  var eventName = triggerDef.event;
+  var shouldPreventDefault = !!triggerDef.preventDefault;
+
+  if (isEnabled('triggersPreventDefault')) {
+    shouldPreventDefault = triggerDef.preventDefault !== false;
+  }
+
+  var shouldStopPropagation = !!triggerDef.stopPropagation;
+
+  if (isEnabled('triggersStopPropagation')) {
+    shouldStopPropagation = triggerDef.stopPropagation !== false;
+  }
+
+  return function (event) {
+    if (shouldPreventDefault) {
+      event.preventDefault();
+    }
+
+    if (shouldStopPropagation) {
+      event.stopPropagation();
+    }
+
+    for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
+      args[_key - 1] = arguments[_key];
+    }
+
+    view.triggerMethod.apply(view, [eventName, view, event].concat(args));
+  };
+}
+
+var ViewEventsMixin = {
+  EventDelegator: EventDelegator,
+  _initViewEvents: function _initViewEvents() {
+    this._domEvents = [];
+  },
+  _undelegateViewEvents: function _undelegateViewEvents() {
+    this.EventDelegator.undelegateAll({
+      events: this._domEvents,
+      rootEl: this.el
+    });
+  },
+  _delegateViewEvents: function _delegateViewEvents() {
+    var view = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : this;
+
+    var uiBindings = this._getUIBindings();
+
+    this._delegateEvents(uiBindings);
+
+    this._delegateTriggers(uiBindings, view);
+  },
+  _delegateEvents: function _delegateEvents(uiBindings) {
+    var _this = this;
+
+    if (!this.events) {
+      return;
+    }
+
+    each(result(this, 'events'), function (handler, key) {
+      if (!isFunction(handler)) {
+        handler = _this[handler];
+      }
+
+      if (!handler) {
+        return;
+      }
+
+      _this._delegate(handler.bind(_this), _this.normalizeUIString(key, uiBindings));
+    });
+  },
+  _delegateTriggers: function _delegateTriggers(uiBindings, view) {
+    var _this2 = this;
+
+    if (!this.triggers) {
+      return;
+    }
+
+    each(result(this, 'triggers'), function (value, key) {
+      _this2._delegate(buildViewTrigger(view, value), _this2.normalizeUIString(key, uiBindings));
+    });
+  },
+  _delegate: function _delegate(handler, key) {
+    var match = key.match(delegateEventSplitter);
+    this.EventDelegator.delegate({
+      eventName: match[1],
+      selector: match[2],
+      handler: handler,
+      events: this._domEvents,
+      rootEl: this.el
+    });
+  }
+};
+
+// DomApi
 
 function setDomApi(mixin) {
-  this.prototype.Dom = _.extend({}, this.prototype.Dom, mixin);
+  this.prototype.Dom = extend$1({}, this.prototype.Dom, mixin);
   return this;
 }
 var DomApi = {
+  // Returns a new HTML DOM node of tagName
```

Comment:
DOM Api was updated to remove jquery by default as `$el` is no longer cached on the view